core/server/common/src/main/java/alluxio/cli/validation/ConfigurationFileParser.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Node;
+import org.w3c.dom.Element;
+
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+
+/**
+ * Parser for configuration files.
+ */
+public class ConfigurationFileParser {
+  /**
+   * Constructs a {@link ConfigurationFileParser} object.
+   */
+  public ConfigurationFileParser() {}
+
+  /**
+   * Parse an xml configuration file into a map.
+   *
+   * Referred to https://www.mkyong.com/java/how-to-read-xml-file-in-java-dom-parser/
+   *
+   * @param path path to the xml file
+   * @return Map from property names to values
+   */
+  public Map<String, String> parseXmlConfiguration(final String path) {
+    File xmlFile;
+    xmlFile = new File(path);
+    if (!xmlFile.exists()) {
+      System.err.format("File %s does not exist.", path);
+      return null;
+    }
+    DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+    DocumentBuilder docBuilder;
+    try {
+      docBuilder = docBuilderFactory.newDocumentBuilder();
+    } catch (ParserConfigurationException e) {
+      return null;
+    }
+    Document doc;
+    try {
+      doc = docBuilder.parse(xmlFile);
+    } catch (IOException e) {
+      return null;
```

Comment:
Print out error message with exception for better user experience.

Reply:
Done.